### PR TITLE
feat(Tag): implement short tag size

### LINKED ISSUE #7512: Short tag
**Component name:** Short tag

_Copied from [previous issue](https://github.com/carbon-design-system/carbon-contribution/issues/23) from 2019 that was not carried to new repo and had design support from Carbon design_

**Why the change is necessary:** 

The Carbon 10 tag is significantly taller (24px) than in Carbon 9 (20px). It does not fit comfortably into the short or compact data tables or the small search field, which means that those elements are unusable if tags need to be incorporated. We are proposing an 18px high tag to allow for sufficient spacing in both situations.

This is especially important now that users can edit their row height on CloudPAL using the new table settings feature.

**Research supporting change:** 

We have heard from our users that they like data density, and that short tables help them scan information faster. During PUP, most users we talked with preferred the short table for resource list, which is why we implemented it. Many users we talk with seem to be using smaller desktop devices. 
 https://ibm.box.com/s/ni1eyykfn6k9k2h66jbzqpsev82c5snc
 
 **Screenshots of proposed design:**

Screen Shot 2021-01-06 at 3 46 22 PM

Screen Shot 2021-01-06 at 3 50 13 PM

**Contributing designer(s) & developer(s):**

Designer: Virginia Nicholson
Developer: Trevor Wong, CloudPAL team

### PR DESCRIPTION
**Pull request by emyarod**
Closes #7512

related #4953

This PR implements support for short Tag and Filter Tag sizes

#### Changelog

**New**

- `size` support for short tags

#### Testing / Reviewing

Confirm the short tag, short filter tag, and short skeleton tags appear correct

### CODE REVIEW

**Review by joshblack — APPROVED** [review]

**Comment by tw15egan** [comment]
Should we reduce the size of the icon in the custom icon variant when the size is small as well?

**Review by laurenmrice — CHANGES_REQUESTED** [review]
- The text in the filterable tag is getting cut off at the bottom for both sizes.
Screen Shot 2021-01-22 at 11 42 38 AM

- Can we name the 18px tag `small` instead of short?
- I don't think we need to change the size of the custom icon when using the small tag. 16px is our smallest icon size and it is also going to be the size of the close icon for the filterable tag still.

**Review by laurenmrice — APPROVED** [review]
Thanks for adding this in! 🙌🏻

**Review by tw15egan — APPROVED** [review]

File: packages/react/src/components/Tag/Tag.Skeleton.js [hunk 1]
```diff
@@ -26,6 +28,12 @@ TagSkeleton.propTypes = {
    * Specify an optional className to add.
    */
   className: PropTypes.string,
+
+  /**
+   * Specify the size of the Tag. Currently supports either `short` or
+   * default sizes.
+   */
+  size: PropTypes.oneOf(['short']),
```
Comment (tw15egan) [inline packages/react/src/components/Tag/Tag.Skeleton.js]: I think this needs to be `sm`

**Review by tw15egan — CHANGES_REQUESTED** [review]

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Pull request merged**